includes/admin/reports/class-wc-admin-report.php

```diff
@@ -322,32 +335,103 @@ public function get_order_report_data( $args = array() ) {
 
 		$query          = apply_filters( 'woocommerce_reports_get_order_report_query', $query );
 		$query          = implode( ' ', $query );
-		$query_hash     = md5( $query_type . $query );
-		$cached_results = get_transient( strtolower( get_class( $this ) ) );
 
 		if ( $debug ) {
 			echo '<pre>';
 			wc_print_r( $query );
 			echo '</pre>';
 		}
+		
+		$query_hash = md5( $query_type . $query );
 
-		if ( $debug || $nocache || false === $cached_results || ! isset( $cached_results[ $query_hash ] ) ) {
-			static $big_selects = false;
-			// Enable big selects for reports, just once for this session
-			if ( ! $big_selects ) {
-				$wpdb->query( 'SET SESSION SQL_BIG_SELECTS=1' );
-				$big_selects = true;
-			}
+		if ( $debug || $nocache || is_null( $result = $this->get_cached_query( $query_hash ) ) ) {
```

Comment:
That result needs to be moved out of the if statement. I don't think phpcs likes this.